File: lib/core/src/main/java/io/loglens/core/ConsoleSink.java
package io.loglens.core;

import java.io.PrintStream;

/**
 * 표준출력 싱크. JVM 단위 테스트, 데스크톱 데모, 그리고 뷰어와의 왕복 검증에 쓴다.
 * 파일 싱크와 <b>같은 접두사 포맷</b>을 낸다 — 파서 하나로 둘 다 읽힌다.
 */
public final class ConsoleSink implements Sink {

    private final PrintStream out;

    public ConsoleSink() { this(System.out); }

    public ConsoleSink(PrintStream out) { this.out = out; }

    @Override
    public void write(Level level, String tag, String body) {
        out.println(LineFormat.render(level, tag, body));
        out.flush();
    }
}

File: lib/core/src/main/kotlin/io/loglens/core/ConsoleSink.kt
package io.loglens.core

import java.io.PrintStream

/**
 * 표준 출력으로 내보냅니다. JVM 테스트, 데스크톱 데모, 뷰어와의 대조 검증에 씁니다.
 * 파일과 **같은 줄 앞부분 형식**을 쓰기 때문에 파서 하나로 둘 다 읽힙니다.
 */
class ConsoleSink @JvmOverloads constructor(
    private val out: PrintStream = System.out,
) : Sink {

    override fun write(level: Level, tag: String, body: String) {
        out.println(LineFormat.render(level, tag, body))
        out.flush()
    }
}
